src/pykeen/utils.py

```diff
@@ -1447,6 +1447,77 @@ def ensure_complex(*xs: torch.Tensor) -> Iterable[torch.Tensor]:
         yield torch.view_as_complex(x)
 
 
+def iter_weisfeiler_lehman(
+    edge_index: torch.LongTensor,
+    max_iter: int,
+    num_nodes: Optional[int] = None,
+) -> Iterable[torch.Tensor]:
+    """
+    Iterate Weisfeiler-Lehman colors.
+
```

Comment:
can you add a bit more narrative about what this is and what it's for? also potentially some references?

Reply:
83b4d28...f006625

Reply:
The WL kernel will be very helpful in analyzing NodePiece features, especially those obtained with the RelationTokenizer when many nodes might be assigned the same hashes.
The WL-test (message passing) will show (if it converges) if it's ever possible to have unique node representations with this strategy. 

tl;dr - think of it as a heuristic whether running a GNN on top of NodePiece features will be useful or not